client/components/forms/form-toggle/index.jsx

```diff
@@ -1,51 +1,72 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classNames = require( 'classnames' );
+import React, { PropTypes } from 'react';
+import PureComponent from 'react-pure-render/component';
+import classNames from 'classnames';
 
-var idNum = 0;
+export default class FormToggle extends PureComponent {
+	static propTypes = {
+		onChange: PropTypes.func,
+		onKeyDown: PropTypes.func,
+		checked: PropTypes.bool,
+		disabled: PropTypes.bool,
+		id: PropTypes.string,
+		className: PropTypes.string,
+		toggling: PropTypes.bool,
+		'aria-label': PropTypes.string,
+		children: PropTypes.node
+	};
 
-module.exports = React.createClass( {
+	static defaultProps = {
+		checked: false,
+		disabled: false
+	};
 
-	displayName: 'FormToggle',
+	static idNum = 0;
 
-	propTypes: {
-		onChange: React.PropTypes.func,
-		checked: React.PropTypes.bool,
-		disabled: React.PropTypes.bool,
-		id: React.PropTypes.string
-	},
+	constructor() {
+		super( ...arguments );
 
-	getDefaultProps: function() {
-		return {
-			checked: false,
-			disabled: false
-		};
-	},
-	_onKeyDown: function( event ) {
-		if ( ! this.props.disabled ) {
-			if ( event.key === 'Enter' || event.key === ' ' ) {
-				event.preventDefault();
-				this.props.onChange();
-			}
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.onClick = this.onClick.bind( this );
+	}
+
+	componentWillMount() {
+		this.id = this.constructor.idNum++;
+	}
+
+	onKeyDown( event ) {
+		if ( this.props.disabled ) {
+			return;
+		}
+
+		if ( event.key === 'Enter' || event.key === ' ' ) {
+			event.preventDefault();
+			this.props.onChange();
 		}
+
```

Comment:
Perhaps in your larger refactor adding some default props for `onChange` and `onKeyDown` could remove some of these conditional checks.